### PR TITLE
sets up bridge controller for upcoming API changes

### PR DESCRIPTION
**Pull request by wilybrace**
* Handles both packed and unpacked cluster detail
* Removes any processing of total_count, used for paging

### CODE REVIEW

File: controllers/dbaas.redhat.com/inventory.go [hunk 1]
```diff
@@ -29,8 +29,8 @@ func (r *CrunchyBridgeInventoryReconciler) discoverInventories(dbaasredhatcomv1a
 		logger.Error(clusterListErr, "Error Listing the instance")
 		return clusterListErr
 	}
-	logger.Info("cluster List ", " Total clusters ", clusterList.Count)
-	if clusterList.Count == 0 {
```
Comment (brandur) [inline controllers/dbaas.redhat.com/inventory.go:33]: +1.

File: internal/bridgeapi/client.go [hunk 2]
```diff
@@ -361,15 +360,23 @@ func (c *Client) ClusterDetail(id string) (ClusterDetail, error) {
 		return ClusterDetail{}, errors.New("unexpected response status from API")
 	}
 
+	// transition type handles both pre-denested and nested unmarshal, leaving
+	// to remaining logic
 	var detail struct {
```
Comment (brandur) [inline internal/bridgeapi/client.go:365]: Ah yeah I see. Okay as mentioned during meeting you don't need this — if you just replaced `var detail struct` with `var detail ClusterDetail` it'll work fine now or later, but it's harmful, so up to you.

**Review by brandur — APPROVED** [review]
Nice! Let's do this.

**Comment by wilybrace** [comment]
Pulled into #35

**Pull request closed without merging**